tests/surrogate_test.py

```diff
@@ -180,6 +180,53 @@ def test_surrogate_1():
     assert numpy.allclose(surrogate_values, exact_values)
 
 
+def test_surrogate_1_multi_input():
+    """Test if surrogate generates exact results when call has >1 input."""
+    grid_gen = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(2))
+    surrogate = Surrogate([(-1, 1), (-1, 1)], grid_gen)
+    surrogate.train(function_1)
+    # random points in the domain
+    points = [(0.649, -0.9), (-0.885, 1)]
+    surrogate_values = [surrogate(x) for x in points]
+    surrogate_values_multi = surrogate(points)
+    assert numpy.allclose(surrogate_values, surrogate_values_multi)
+
+
+def test_surrogate_1_multi_grid():
+    """Test if surrogate generates exact results if call has >1 input."""
+    grid_gen = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(2))
+    surrogate = Surrogate([(-1, 1), (-1, 1)], grid_gen)
+    surrogate.train(function_1)
+    # random points in the domain
+    points = numpy.random.rand(3,4,2)*2 - 1
+    exact_values = numpy.zeros(points.shape[:-1])
+    for i in range(points.shape[0]):
+        for j in range(points.shape[1]):
+            exact_values[i,j] = function_1(tuple(points[i,j]))
+    surrogate_values_multi = surrogate(points)
+    assert numpy.allclose(exact_values, surrogate_values_multi)
+
+
+def test_surrogate_2D_multi_shape():
+    """Test if surrogate call generates the correct shape for 2 dimensions."""
+    grid_gen = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(2))
+    surrogate = Surrogate([(-1, 1),(-1, 1)], grid_gen)
+    surrogate.train(function_1)
+    # random points in the domain
+    point_scalar = [1, 1]
+    point_nested_list = [[1, 1]]
+    points_list = [[1, 1], [0.5, 0.5]]
+    point_grid = numpy.ones((1,1,1,2))
+    points_grid_1 = numpy.ones((4,3,2,2))
+    points_grid_2 = numpy.ones((4,3,5,2))
+    assert numpy.isscalar(surrogate(point_scalar))
+    assert surrogate(point_nested_list).shape == (1,)
+    assert surrogate(points_list).shape == numpy.shape(points_list)[:-1]
+    assert surrogate(point_grid).shape == (1,1,1)
+    assert surrogate(points_grid_1).shape == (4,3,2)
+    assert surrogate(points_grid_2).shape == (4,3,5)
```

Comment:
These tests will be easier to read if you just put the variables inside the `surrogate(...)`. Also on line 224, hard code the expected shape, like the others.